routers/private/hook.go

```diff
@@ -119,6 +120,48 @@ func HookPreReceive(ctx *macaron.Context) {
 			})
 			return
 		}
+
+		// check signed commits
+		if protectBranch.RequireSignedCommits {
+			gitRepo, err := git.OpenRepository(repo.RepoPath())
+			if err != nil {
+				log.Error("Unable to find the git repository %s/%s: %v", ownerName, repoName, err)
+				ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
+					"err": fmt.Sprintf("Unable to find the git repository %s/%s", ownerName, repoName),
+				})
+				return
+			}
+
+			stdout, err := git.NewCommand("rev-list", oldCommitID+"..."+newCommitID).RunInDirBytes(repo.RepoPath())
```

Comment:
In pre-receive the new commits and objects are not placed directly into the object tree (depending on your git version of course). They are placed in to Quarantine and/or various other places.

You need to set the environment for any git commands to tell it to look in those directories for objects too.

https://github.com/lunny/gitea/blob/b39e13d26d6caf503d17563ee9c4d9917570e169/routers/private/hook.go#L68-L80